x-pack/plugins/security_solution/server/lib/entity_analytics/entity_store/saved_object/engine_descriptor_type.ts

```diff
@@ -55,11 +58,33 @@ const version1: SavedObjectsModelVersion = {
   ],
 };
 
+const version2: SavedObjectsModelVersion = {
+  changes: [
+    {
+      type: 'mappings_addition',
+      addedMappings: {
+        error: { type: 'object' },
+      },
+    },
+    {
+      type: 'data_backfill',
+      backfillFn: (document) => {
```

Comment:
If you are adding a new default field to the schema that is optional we don't need to run this backfill operation, leaving it `undefined` should be fine. Or does `null` mean something specific in this case?

Reply:
no, it was meant to clearly state there was no value as its optional. I wasn't sure whether just leaving it undefined would work. I'll update